src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_postgres.py

```diff
@@ -345,6 +346,53 @@ def flexible_server_connection_string(
     }
 
 
+def connect_to_flexible_server_postgresql(cmd, server_name, administrator_login, administrator_login_password, database_name=None, postgres_query=None):
+    host = '{}.postgres.database.azure.com'.format(server_name)
+    cursor = None
+    json_data = None
+    if database_name is None:
+        logger.warning("Connecting to postgres database by default.")
+        database_name = "postgres"
+    # Connect to mysql and get cursor to run sql commands
+    try:
+        connection_kwargs = {
+            'host': host,
+            'database': database_name,
+            'user': administrator_login,
+            'password': administrator_login_password
+        }
+        connection = psycopg2.connect(**connection_kwargs)
+        connection.set_session(autocommit=True)
+        logger.warning('Successfully connected to PostgreSQL.')
+    except Exception as e:
+        logger.warning('Failed connection to PostgreSQL. Check error and validate firewall and public access settings.')
+        raise CLIError("Unable to connect to PostgreSQL Server: {}".format(e))
```

Comment:
could we consider reuse the code her https://github.com/Azure/azure-cli-extensions/blob/master/src/db-up/azext_db_up/custom.py#L366-L394?

I mean move this part to your module and reuse them.

Reply:
Yes, used db-up module as a reference point for this code.  We used the basic connection part of the code, but didn't need to set user permissions.  Also added code to run a simple query against the server for customer to test connection.